src/main/java/com/github/dockerjava/core/util/DockerImageName.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright 2015 jgriffiths.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.dockerjava.core.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Written to construct and validate docker image names, and their respective parts:
+ * [REGISTRY/][NAMESPACE/]REPOSITORY[:TAG]
+ * <p>
+ * Based on Docker v1.5 source
+ *
+ * @author jgriffiths 08/04/2015
+ */
+public class DockerImageName {
+
+    public String imageName;
+
+    private TagNamePart tag = null;
+    private RepositoryNamePart repository = null;
+    private NamespaceNamePart namespace = null;
+    private RegistryNamePart registry = null;
```

Comment:
Sure, thanks. Just trying to make any discussion. Still think that simple regexp may solve all parsing...